ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -310,11 +310,27 @@ private static ConnectSchema toConnectSchema(
     private final Set<ColumnName> seenKeys = new HashSet<>();
     private final Set<ColumnName> seenValues = new HashSet<>();
 
+    private boolean failOnDuplicates = true;
+
     public Builder withRowTime() {
       columns.add(IMPLICIT_TIME_COLUMN);
       return this;
     }
 
+    /**
+     * Allows key and value columns with the same name.
+     *
+     * <p>This should only be used when building the schemas used by Kafka Streams, where
+     * the key columns are copied into the value schema. KSQL does not support duplicate column
+     * names in data sources.
+     *
+     * @return self.
+     */
+    public Builder allowDuplicates() {
```

Comment:
as discussed offline, I think having this makes things a little more confusing. If we wanted to make it strongly typed, I think we should have a different class that allows duplicates but I think that's overkill

Reply:
Yep, cool, my thinking was going down the same route.